src/runtime-rs/config/configuration-dragonball.toml.in

```diff
@@ -318,3 +318,12 @@ static_sandbox_resource_mgmt=@DEFSTATICRESOURCEMGMT_DB@
 # - "/path/to:ro", readonly mode.
 # - "/path/to:rw", readwrite mode.
 sandbox_bind_mounts=@DEFBINDMOUNTS@
+
+# Base directory of directly attachable network config.
+# Network devices for VM-based containers are allowed to be placed in the
+# host netns to eliminate as many hops as possible, which is what we
+# called a "Directly Attachable Network". The config, set by special CNI
+# plugins, is used to tell the Kata containers what devices are attached
+# to the hypervisor.
+# (default: /run/kata-containers/dans)
```

Comment:
Is there any reason we have to use the path `/run`, I think most of the time we use `/etc` for configuration.

Reply:
`/etc` is for static configs, like configs for Kata Containers. However, this is a runtime config which is managed by CNI plugins. The lifecircle is same as pod. So that's why I prefer to use `/run`.

Reply:
I see. But if this config is managed by the CNI plugins, I assume the users don't need to configure it in the toml file, maybe it's unnecessary to add it in the `configuration.toml`, wdyt ?

Reply:
The best way is passed this path from annotations. The path is transparent to the Kata.

As of today, the CNI only pass a netns path (the DAN not depends on netns), which is leading that the plugins is not able to pass extra data to the runtime. A temporaroy solution is to write network configs into a file, whose the path is  pre-negotiated by Kata runtime and CNI plugins.

The runtime **MUST** know the path, and the path could be modified if the CNI plugins write to a new place (pre-negotiatation). Kata Containers only read from those files and setup network. The CNI plugins are responsible for writing and managing those files.

Reply:
The attachable network config file will be generated by cni and will be used by runtime-rs. Do I understand it correctly?

Reply:
Exactly.

Reply:
And although CNI still needs to configure `eth0` in container netns, it actually doesn't have any effect, right?

Reply:
Where does the `eth0` come from?

- DAN enabled: the Network config is loaded from a DAN config. Further, if the config contains a device named `eth0`, the `eth0` device is still available for guest.
- DAN disabled (normal way): the Kata runtime scans a netns for network devices. Further, if a device named `eth0` is placed in the netns, the `eth0` device is still available for guest.

Reply:
What does CNI do in pod netns when DAN enabled? AFAIK, it must create an `eth0` in pod netns.

Reply:
You might get rid of knowledge of existed CNI plugins. Adding a network device named `eth0` into a pod netns is one of things we have no alternative but to do, since some runtimes, such as runc, aren't able to find network devices dynamically. The DAN gives the Kata Containers a flexible way to discover networks.

For example, a CNI that supports the DAN feature could setup a tap device named `hosttap0` in the host netns. The plugin writes the tap device into `/path/to/dan/config.json`. Then, a Kata Containers is launched and found the config existed, the Kata attachs that tap device into pod and won't scan the netns anymore. At this time, networking for the pod is setup successfully.

If the plugin want to be compatiable with runc containers, this is a bit of a hassle. The plugin still needs to create a netns and setup a device named `eth0` (a veth device in most cases) for runc. In addition, the plugin should write a DAN config and setup a host tap device for Kata.

Reply:
gotcha. This feature cannot yet be used in k8s with kata runtime.

Reply:
You can use it once a specific CNI plugin supported DAN is released.